File: 백준/Silver/2839. 설탕 배달/설탕 배달.c
#include <stdio.h>

int main()
{
    int i, j, N = 0, tmp = 0;
    scanf("%d", &N);
    tmp = N;
    for (i = 0; i < N; i++)
    {
        for (j = 0; j < N; j++)
            if (N == j * 5 + i * 3)
            {
                if (tmp > j + i)
                    tmp = j + i;
            }
    }
    if (tmp == N)
        printf("-1");
    else
        printf("%d", tmp);
}

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int N = scan.nextInt();
        int tmp = N;
        
        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++)
                if (N == j * 5 + i * 3) {
                    if (tmp > j + i)
                        tmp = j + i;
                }
        }
        
        if (tmp == N)
            System.out.println("-1");
        else
            System.out.println(tmp);
    }
}